src/NServiceBus.Core/Sagas/ActiveSagaInstance.cs

```diff
@@ -222,9 +210,22 @@ static object GetDefault(Type type)
 
         internal class CorrelationPropertyInfo
         {
-            public bool HasExistingValue;
-            public PropertyInfo PropertyInfo;
-            public object Value;
+            public CorrelationPropertyInfo(PropertyInfo propertyInfo, IContainSagaData sagaData)
+            {
+                this.sagaData = sagaData;
+
+                PropertyInfo = propertyInfo;
+                InitialValue = propertyInfo.GetValue(sagaData);
+                var defaultValue = GetDefault(propertyInfo.PropertyType);
+                HasInitialValue = InitialValue != null && !InitialValue.Equals(defaultValue);
+            }
+
+            IContainSagaData sagaData;
+
+            public bool HasInitialValue { get; }
+            public PropertyInfo PropertyInfo { get; }
+            public object InitialValue { get; }
+            public object CurrentValue => PropertyInfo.GetValue(sagaData);
```

Comment:
I'm OK with changing it to a method. Although I would rather prefer to pass the value around. But not hung up on the approach